lib/dry/transaction/step.rb

```diff
@@ -13,24 +13,26 @@ class Step
       attr_reader :operation_name
       attr_reader :operation
       attr_reader :options
+      attr_reader :block
       attr_reader :call_args
 
-      def initialize(step_adapter, step_name, operation_name, operation, options, call_args = [])
+      def initialize(step_adapter, step_name, operation_name, operation, options, call_args = [], &block)
         @step_adapter = step_adapter
         @step_name = step_name
         @operation_name = operation_name
         @operation = operation
         @options = options
+        @block = block
         @call_args = call_args
       end
 
-      def with_call_args(*call_args)
-        self.class.new(step_adapter, step_name, operation_name, operation, options, call_args)
+      def with_call_args(*call_args, &block)
```

Comment:
Yes, sure, @timriley, that was a matter of half-blind extending everything with blocks :)

I will remove it.